students/Jaidjen/lesson05/Mailroom3.py

```diff
@@ -0,0 +1,80 @@
+import sys  # imports go at the top of the file
+
+
+Mail = {1:{'Name':'James Smith', 'Donation':[2000.25,500,27.50]},
+              2:{'Name': 'Dorothy Parker','Donation':[3175,475.45,92.30]},
+              3:{'Name':'Dohgyu Hwang','Donation':[10670,350.19,175]},
+              4:{'Name': 'Sally Yu', 'Donation':[100,925]},
+              5:{'Name':'Jenny Ramos','Donation':[25.14]}}
+
+prompt = "\n".join(("Functions:",
+          "Please choose from below options:",
+          "1 - Send a Thank You to a single donor",
+          "2 - Create a Report",
+          "3 - Send letters to all donors",
+          "4 - Quit",
+          ">>> "))
+
+
+def view_mail(Mail):
+    print("{:20s}{:20s}{:20s}{:20s}".format("DONOR ", "| TOTAL GIVEN ", "| NUMBER OF GIFTS ", "| AVERAGE GIFTS "))
+    print("-----------------------------------------------------------------------------")
+    for mail in Mail:
+        print("{:20s}{:^20.2f}{:^19d}{:^18.2f}".format(Mail[mail]['Name'],sum(Mail[mail]['Donation']), len(Mail[mail]['Donation']),(sum(Mail[mail]['Donation'])/(len(Mail[mail]['Donation'])))))
+
+
+def add_mail():
+    print("If you want a list of donors please type 'list' ")
+    new_mail = input("Please enter the full name of the donor? ")
+    if new_mail == 'list':
+        print("{:<20s}".format(Mail[Name]))
+    else:
+        try:
+            new_mail == Mail[1]
+            dnr_amt = input("Please enter the donation amount: ")
+        except ValueError:
+            print('Not acceptable')
+            raise ValueError
```

Comment:
if you're catching the ValueError why raise it again? best to keep asking the user until valid value is entered.